pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -175,6 +175,9 @@ type XdsConnection struct {
 	RouteConfigs  map[string]*xdsapi.RouteConfiguration `json:"-"`
 	HTTPClusters  []*xdsapi.Cluster
 
+	// Last nonce sent and ack'd (timestamps) used for debugging
+	NonceSent, NonceAcked string
+
```

Comment:
This is not right.. Nonces are specific to the resource type. i.e. in a given ADS stream, you could have separate nonces for every route, one for all clusters, one for all listeners, and separate nonce for every set of endpoints (i.e. one per cluster). So, you might have to have a table somewhere, of the resources sent out in a response (list of routes by name, then an entry for all clusters, then another for all listeners, and then a list of cluster names [for endpoints]. Then, upon receiving the discovery requests, you check this table (based on requested resource type) to see if any previous nonce has been acked).

This would mean that the thing you show on the CLI has got to be divided further into sort of status flags (RDS(tick), LDS(tick), CDS(tick), EDS(tick)). And since the nonces are changing everytime you send out a new resource, you would have to garbage collect old ones.